go/def.bzl

```diff
@@ -431,6 +431,17 @@ def _emit_go_link_action(ctx, importmap, transitive_libs, cgo_deps, lib,
     if d.basename.endswith('.so'):
       dirname = _short_path(d)[:-len(d.basename)]
       ldflags += ["-Wl,-rpath,$ORIGIN/" + ("../" * pkg_depth) + dirname]
+  filtered_gc_linkopts = []
+  is_extldflags = False
```

Comment:
whenever I see this sort of loop-surviving boolean, it screams "factor out a method" to me.